R/plotPropselRange.R

```diff
@@ -0,0 +1,225 @@
+#' @importFrom graphics lines
+NULL
+
+#' Plot classification accuracy indices at different proportions of selection
+#'   or at different threshold (cutoff) values
+#' 
+#' \code{plotPropselRange} plots classification accuracy indices at different
+#' proportions of selection under partial and strict invariance conditions for 
+#' a given CFA fit.
+#' 
+#' @param cfa_fit CFA model output from lavaan.
+#' @param pmix List of length `g` containing the mixing proportions of each
+#'     group (where `g` is the number of groups). If `NULL`, defaults to `1/g` 
+#'     for each group (i.e., the populations have equal size).
+#' @param labels A character vector with `g` elements to label the reference
+#'     and focal groups on the plot, where `g` is the number of groups. If not
+#'     provided, groups are labeled automatically as 'Reference' (for the first
+#'     group) and 'Focal_1' through 'Focal_(g-1)', where `g` is the number of
+#'     groups.
+#' @param cai_names A vector of strings indicating the classification accuracy
+#'     indices of interest. c("PS", "SR", "SE", "SP") by default.
+#' @param mod_names A vector of strings indicating the invariance conditions of
+#'     interest. c("par", "str") by default.
+#' @param from The lowest proportion of selection to consider. `0.01` by default.
+#' @param to The largest proportion of selection to consider. `0.25` by default.
+#' @param by The increment of the sequence of proportions. `0.01` by default.
+#' @param cutoffs_from The lowest threshold to consider.`NULL` by default.
+#' @param cutoffs_to The largest threshold to consider. `NULL` by default.
+#' @param plotAIs Whether a plot for AIs should be printed. `TRUE` by default.
+#' @return Eight plots illustrating how proportion selected (PS), success ratio 
+#'     (SR), sensitivity (SE), and specificity (SP) change across different 
+#'     proportions of selection under partial and strict invariance conditions.
+#' @examples
+#' \dontrun{
+#' library(lavaan)
+#' HS <- HolzingerSwineford1939
+#' HS$sex <- as.factor(HS$sex)
+#' HS.model <- ' visual  =~ x1 + x2 + x3
+#'               textual =~ x4 + x5 + x6
+#'               speed   =~ x7 + x8 + x9 '
+#' 
+#' fit <- cfa(HS.model, data = HS, group = "sex")
+#' 
+#' # plot CAI at different proportions of selection
+#' plotPropselRange(fit, pmix = table(HS$sex)/sum(table(HS$sex)))
+#' 
+#' # plot CAI at different cutoffs
+#' plotPropselRange(fit, pmix = table(HS$sex)/sum(table(HS$sex)), 
+#'    cutoffs_from = 35, cutoffs_to = 50)
+#'
+#' # plot only SR under partial invariance for up to 10% selection.
+#' plotPropselRange(fit, pmix = table(HS$sex)/sum(table(HS$sex)), 
+#'     from = 0.01, to = 0.10, cai_names = "SR", mod_names = "par")
```

Comment:
Since AI is a slightly different index and the setup for it is separate within the function, I added a `plotAIs` parameter which is `TRUE` by default, but the user can indicate that they do not wish for the AI plot to be printed. Is this alright?